src/event-log/event-log.ts

```diff
@@ -0,0 +1,32 @@
+export type Event = {
+  watermark: string,
+  messageCid: string
+};
+
+export interface EventLog {
+ /**
+  * opens a connection to the underlying store
+  */
+  open(): Promise<void>;
+
+  /**
+   * closes the connection to the underlying store
+   */
+  close(): Promise<void>;
+
+  /**
+   * adds an event to a tenant's event log
+   * @param tenant - the tenant's DID
+   * @param messageCid - the CID of the message
+   */
+  append(tenant: string, messageCid: string): Promise<string>
+
+  /**
+   * retrieves all of a tenant's events that occurred after the watermark provided.
+   * If no watermark is provided, all events for a given tenant will be returned.
+   *
+   * @param tenant
+   * @param watermark
+   */
+  getEventsAfter(tenant: string, watermark?: string): Promise<Array<Event>>
```

Comment:
Question: (fine if this is out of scope) have you thought about how we handle large amount events to be returned?

An easy/lazy way to do this would be to wrap `Event[]` and return a subset if exceeding some threshold:

```javascript
{
  events: Event[];
  hasMore: boolean;
}
```

Reply:
definitely crossed my mind. am going to hold off until we discuss how we want to approach pagination in general